scio-core/src/main/scala/com/spotify/scio/JavaConverters.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio
+
+import org.apache.beam.sdk.io.{DefaultFilenamePolicy, FileBasedSink}
+import org.apache.beam.sdk.io.fs.ResourceId
+import org.apache.beam.sdk.options.ValueProvider.StaticValueProvider
+
+object JavaConverters {
+
+  implicit def toResourceId(filenamePrefix: String): ResourceId =
```

Comment:
IMHO we should avoid using implicit type conversion. Using implicit class is a better approach.

Reply:
👍 yes that would nicer

Reply:
In other words?

```scala
implicit class RichString (x: String) {
  def asResourceId(): ResourceId = FileBasedSink.convertToFileResourceIfPossible(x)
  def asFilenamePolicy(): DefaultFilenamePolicy = DefaultFilenamePolicy.fromStandardParameters(
      StaticValueProvider.of(x),
      null,
      null,
      false)
}
```

Reply:
LGTM